salt/modules/groupadd.py

```diff
@@ -123,41 +165,74 @@ def getent(refresh=False):
         return __context__['group.getent']
 
     ret = []
-    for grinfo in grp.getgrall():
+    if root is not None:
+        _fun = functools.partial(_getgrall, root=root)
```

Comment:
Even GvR himself [doesn't agree](https://twitter.com/gvanrossum/status/391769557758521345?lang=en) with you here! :laughing: Me too, BTW.

Reply:
Can I disagree here? The example is for partials that do not expect arguments (is done explicit two or three comments later), and this is not the case. I find this more explicit than this:

```python
    if root is not None:
        _fun = lambda name: _getgrnam(name, root=root)
    else:
        _fun = lambda name: grp.getgrnam(name)
```

In the lambda form is not explicit that the call is not evaluated (lazy evaluation), and that the free parameter is indeed name.

If you have an strong objection I can change it.

```

Reply:
Umh. E731 (do not assign a lambda expression, use a def) happens in this case.

I like `lambda`s as anyone, but in this case if some new user see the code, she/he will find documentation about [functools.partial](https://docs.python.org/3.7/library/functools.html#functools.partial) easily.

Tangentially about this topic, is there any set of WXXX or EXXX that the team agrees to ignore?

Reply:
Ahh, you're [running into this](https://www.python.org/dev/peps/pep-0008/#programming-recommendations).

```python
if root:
    def _fun(name): return _getgrnam(name, root=root)
else:
    def _fun(name): return grp.getgrnam(name)
```

Reply:
:confused:
What how is defining a local inner function be better than a standard library that is documented? Is there any plan to drop `functools.partial` that I am not aware of?

Also is used along the codebase of salt too:

* https://github.com/saltstack/salt/blob/develop/salt/utils/parsers.py#L24
* https://github.com/saltstack/salt/blob/develop/salt/utils/boto3mod.py#L40
* https://github.com/saltstack/salt/blob/develop/salt/utils/botomod.py#L40
* https://github.com/saltstack/salt/blob/develop/salt/utils/roster_matcher.py#L81
* https://github.com/saltstack/salt/blob/develop/salt/modules/rpmbuild.py#L557
...

Reply:
@aplanas I am just fixing you a lint error according to as [PEP-8 says](https://www.python.org/dev/peps/pep-0008/#programming-recommendations). I still find this:

```python
def foo(x): f(x)
```

Much better than this:

```python
foo = functools.partial(f, x)
```

Reply:
For future reference. I found the pep about currying:

https://www.python.org/dev/peps/pep-0309/